packages/expect-utils/src/utils.ts

```diff
@@ -390,12 +390,16 @@ export const arrayBufferEquality = (
   a: unknown,
   b: unknown,
 ): boolean | undefined => {
-  if (!(a instanceof ArrayBuffer) || !(b instanceof ArrayBuffer)) {
-    return undefined;
-  }
+  let dataViewA = a as DataView;
+  let dataViewB = b as DataView;
 
-  const dataViewA = new DataView(a);
-  const dataViewB = new DataView(b);
+  if (!(a instanceof DataView && b instanceof DataView)) {
+    if (!(a instanceof ArrayBuffer) || !(b instanceof ArrayBuffer))
+      return undefined;
+
+    dataViewA = new DataView(a);
+    dataViewB = new DataView(b);
+  }
```

Comment:
Perhaps simply like this (just a quick idea):

```ts
  let dataViewA: DataView;
  let dataViewB: DataView;

  if (a instanceof DataView && b instanceof DataView) {
    dataViewA = a;
    dataViewB = b;
  } else if (a instanceof ArrayBuffer && b instanceof ArrayBuffer) {
    dataViewA = new DataView(a);
    dataViewB = new DataView(b);
  } else {
    return undefined;
  }
```

Reply:
Thank you, I've reimplemented it. What do you think?